### DISCUSSION TITLE
Company autocompletion like YCM

### DISCUSSION THREAD

**Discussion by ChoppinBlockParty** [company-mode/company-mode#1193] [Q&A] [answered]
Hi,
Have read [this pull](https://github.com/company-mode/company-mode/pull/524) and [this wiki](https://github.com/company-mode/company-mode/wiki/Switching-from-AC), however nothing from there is working. Can anyone shed the light on what is the last state of this feature? How it can be used to achieve YCM-like TAB autocompletion?

**Comment by dgutov** [comment]
What exactly isn't working?

**Comment by ChoppinBlockParty** [comment]
It is not working like TAB in YCM: (1) I am typing, (2) completion menu pops up, (3) I press TAB (mb several times), (4) a completion item is selected in the popup, I can see it in my line, but the completing part of the item is right from the cursor, **if I continue typing the completing part disappears** (here I expect to continue typing after the completing part of the word, i.e. completing part is preserved after pressing TAB, like in YCM).

**Comment by dgutov** [comment, marked as answer]
You can try https://github.com/company-mode/company-mode/blob/master/company-tng.el instead.

It allows a different workflow, probably closer to your expectations.

**Comment by Alexander-Shukaev** [comment]
@ChoppinBlockParty, were you satisfied with `company-tng`?

**Comment by dgutov** [comment]
Any troubles with `company-tng` should be turned into separate reports.
